Automattic/pocket-casts-android
# Fix/issue 1228 theme selection UI bug

## Description

This PR fixes an issue where when a user is not logged into a PC Plus account and selects a Plus theme, the UI will indicate the wrong theme as being selected once the user exits the Plus on boarding screen. Furthermore, the UI would scroll back to the "selected" theme even though that was not the actual active theme.

Fixes #1228

## Testing Instructions




1. With a Free account
2. Go to "Profile" → ⚙️ → "Appearance"
3. Select a "Free" theme
4. The app should update with the selected theme and the UI should indicate the selected theme.
5. Select a "Plus" theme
6. The on boarding screen should come up
7. Navigate back to the theme selection view
8. The app should be in the last selected "Free" theme and the UI should indicate that said theme is selected

## Screenshots or Screencast 

#### Before:
Check the original issue for a Before screen cap of the issue.
#### After:
https://github.com/Automattic/pocket-casts-android/assets/68164093/ec5046ca-9311-4647-a480-e2679220e862


## Checklist
   - No strings added
   - No Jetpack compose components added or changed.
 
#### I have tested any UI changes...


## Notes
I tested my changes without lines [126-127](https://github.com/P-Burk/pocket-casts-android/blob/6639c81f10be103486bf0738d3da9cdf81a8d197/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceSettingsFragment.kt#L126) as well as with those two lines still in the code, and the bug fix performs as expected/desired either way (for me at least). I left these two lines in because I didn't want to delete something that I don't feel like I fully understand why it's there in the first place.
